src/vector.rs

```diff
@@ -571,4 +585,33 @@ mod tests {
         assert_nearly_eq!(expected_projection.x, projection.x);
         assert_nearly_eq!(expected_projection.y, projection.y);
     }
+
+    #[test]
+    fn vector_rotated_by_0_does_not_change() {
+        let vector = Vector { x: 5.0, y: 10.0 };
+        let rotated_vector = vector.rotate(Radians::try_new(0.0).unwrap());
+
+        assert_nearly_eq!(vector.x, rotated_vector.x, 0.00001);
+        assert_nearly_eq!(vector.y, rotated_vector.y, 0.00001);
+    }
+
+    #[test]
+    fn vector_rotated_by_pi_is_correct() {
+        let vector = Vector { x: 5.0, y: 10.0 };
+        let rotated_vector = vector.rotate(Radians::try_new(PI).unwrap());
+
+        let expected_vector = Vector { x: -5.0, y: -10.0 };
+        assert_nearly_eq!(expected_vector.x, rotated_vector.x, 0.00001);
+        assert_nearly_eq!(expected_vector.y, rotated_vector.y, 0.00001);
+    }
+
+    #[test]
+    fn vector_rotated_by_half_pi_is_correct() {
+        let vector = Vector { x: 5.0, y: 10.0 };
+        let rotated_vector = vector.rotate(Radians::try_new(FRAC_PI_2).unwrap());
+
+        let expected_vector = Vector { x: -10.0, y: 5.0 };
+        assert_nearly_eq!(expected_vector.x, rotated_vector.x, 0.00001);
+        assert_nearly_eq!(expected_vector.y, rotated_vector.y, 0.00001);
+    }
 }
```

Comment:
Missing test cases: 
- Rotating by 2π and -2π
- Rotating twice by π
- Rotation in clockwise direction

Reply:
230a03c

Reply:
Note: Radians are only in the range 0 <= x < 2π

Reply:
Remove code duplication and improve precision